pkg/lambda/options.go

```diff
@@ -0,0 +1,105 @@
+package lambda
+
+import (
+	"errors"
+	"os"
+	"path"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/pflag"
+	"go.uber.org/multierr"
+
+	"github.com/atlassian/gostatsd/internal/awslambda/extension/api"
+	"github.com/atlassian/gostatsd/pkg/statsd"
+)
+
+const (
+	ParamRuntimeAPI        = `lambda-runtime-api`
+	ParamEntryPointName    = `lamda-entrypoint-name`
+	ParamEnableManualFlush = `enable-manual-flush`
+	ParamTelemetryAddr     = `telemetry-addr`
+)
+
+// Options is used to provide overrides when calling `NewExtension`
+type Options struct {
+	// Server is extended by the extension
+	// This value is required, no defaults are provided.
```

Comment:
If things are required, they should be passed as parameters.  The struct should be optional stuff only, if it's intention is to only be overridden sometimes.